### PR TITLE
Update lint rake task for CircleCI deploys

### PR DESCRIPTION
**Pull request by Bodacious**
Lint rake task will now run Rubocop against a list of all Ruby files that have been edited between the HEAD branch and the base.

By default, base branch is 'develop', but this can be specified like so:

``` bash
rake lint:rubocop RUBOCOP_LINT_BASE=my-other-base
```

Have a look at the regex in [lib/tasks/lint.rake](https://github.com/boxt/boxt_ruby_style_guide/pull/21/files#diff-9fa751bfbcf351f423b25e0abe8bf8d8) to make sure you agree with the files included for linting.

### CODE REVIEW

File: lib/tasks/lint.rake [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# The base branch to compare HEAD with for changes
+BASE_BRANCH = "develop"
+
+##
+# Directories containing Ruby files to lint
+RUBY_DIRS = %w[app lib test spec].freeze
+
+##
+# Grep file pattern to make sure we only check files that are .rb
+GREP_PATTERN = "'(#{RUBY_DIRS.join('|')})\/.+\.(rb|rake)'"
```
Comment (schinery) [inline lib/tasks/lint.rake:16]: Can we include `Gemfile` and `.gemspec` in this pattern? They are covered by the current cops/rules so would be good to include them too.

**Comment by Bodacious** [comment]
Sure

Do we want to include any other files too? (I felt it best to exclude things like `schema.rb` and migration files. Which might not be worth the effort of linting.

File: lib/tasks/lint.rake (hunk 1 above)
Comment (Bodacious) [reply lib/tasks/lint.rake:16]: Sure

Do we want to include any other files too? (I felt it best to exclude things like `schema.rb` and migration files. Which might not be worth the effort of linting.

Comment (schinery) [reply lib/tasks/lint.rake:16]: Yeah defo not the schema. Currently, the migrations are covered by cops too but maybe that is overkill. Maybe just the two mentioned for now, we can always tweak things if we discover other types later.

Comment (Bodacious) [reply lib/tasks/lint.rake:16]: Added Rakefile too

**Comment by Bodacious** [comment]
I've updated the `.circleci/config.yml` to use the revision base from the pull request (if present). For this to work, I had to update the CircleCI API version from 2 to 2.1

**Review by michaelbulpitt — APPROVED** [review]
LGTM! 🥳

**Review by schinery — APPROVED** [review]
LGTM. As mentioned I would merge straight to master and we'll update projects to use the latest version

**Pull request merged**